### PR TITLE
perf(es/parser): Use smarter lookup table for lexer

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

`phf` is slower than expected.

**Related issue (if exists):**

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_parser

**Comment by kdy1** [comment]
# Performance diff

```

es/parser/ts.        time:   [142.47 ms 142.73 ms 143.04 ms]
                        change: [-4.6391% -4.3986% -4.1497%] (p = 0.00 < 0.05)
```

File: crates/swc_ecma_parser/src/lexer/table.rs [hunk 1]
```diff
@@ -59,8 +59,230 @@ const ERR: ByteHandler = Some(|lexer| {
...
+
+const L_S: ByteHandler = Some(|lexer| {
+    lexer.read_word_with(|s| match s {
+        "super" => Some(Word::Keyword(Keyword::Super)),
+        "static" => Some(Word::Ident(IdentLike::Known(KnownIdent::Static))),
+        "switch" => Some(Word::Keyword(Keyword::Switch)),
+        "symbol" => Some(Word::Ident(IdentLike::Known(KnownIdent::Symbol))),
+        "set" => Some(Word::Ident(IdentLike::Known(KnownIdent::Set))),
+        "string" => Some(Word::Ident(IdentLike::Known(KnownIdent::String))),
+        "satisfies" => Some(Word::Ident(IdentLike::Known(KnownIdent::Satisfies))),
```
Comment (magic-akari) [inline crates/swc_ecma_parser/src/lexer/table.rs:232]: Some TypeScript keywords are also included. We did not do this before.
Is that okay?

Comment (kdy1) [reply crates/swc_ecma_parser/src/lexer/table.rs:232]: It's intended. It allows skipping an access to `phf::Map` from https://github.com/swc-project/swc/pull/8226/files#diff-6402bc52db5ddbd503b09a01f2ab9ec09c93953a43f40d641aa0edfa83f0cf08R789

We don't check for `KnownIdent`

**Review by magic-akari — APPROVED** [review]

**Pull request merged**